File: 724-FindPivotIndex/724-FindPivotIndex.php
class Solution {
// 25 min
    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function pivotIndex($nums) {
        $sumLeft = 0;
        $sumRight = array_sum($nums) - $nums[0];
        if ($sumLeft === $sumRight) {
            return 0;
        }
        for ($i = 1; $i < count($nums); $i++) {
            $sumLeft += $nums[$i-1];
            $sumRight -= $nums[$i];

            if ($sumLeft === $sumRight) {
                return $i;
            }
        }

        return -1;
    }
}

File: 724-FindPivotIndex/724-FindPivotIndex.py
# Last updated: 18/04/2025, 14:17:51
class Solution(object):
    def pivotIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        sum1 = 0
        sum2 = sum(nums)
        prev = 0
        for key, num in enumerate(nums):
            sum1 += prev
            sum2 -= num

            if sum1 == sum2:
                return key
            prev = num
        
        return -1
        